torch/_dynamo/variables/torch.py

```diff
@@ -144,6 +144,75 @@ def _dynamo_overriden_transformers_eq(self, other):
     pass
 
 
+class TorchCtxManagerClassVariable(VariableTracker):
+    def __init__(self, value, **kwargs):
+        super().__init__(**kwargs)
+        self.value = value
+
+    def unique_var_name(self):
+        name = torch_get_name(self.value, f"allowed_fn_{id(self.value)}")
+        return "__" + re.sub(r"[^a-zA-Z0-9_]+", "_", name)
+
+    def reconstruct(self, codegen):
+        return codegen.setup_globally_cached(self.unique_var_name(), self.value, False)
```

Comment:
This is duped from TorchVariable right?

Reply:
Yes, will refactor out a shared function.